.github/workflows/deploy.yml

```
@@ -1,103 +1,168 @@
+        uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4
+        with:
+          node-version: '24'
+      - name: Check ONNX models are real binaries on HuggingFace CDN
+        run: node scripts/validate-onnx-models.js
+
+
   deploy-preview:
     name: Deploy Preview
-    needs: [lint-test, validate]
+    needs: [lint-test, validate, smoke-test, validate-models]
     if: github.event_name == 'pull_request' && secrets.VERCEL_TOKEN != ''
```

actionlint (1.7.12)

[error] context "secrets" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)